arrow.go

```diff
@@ -93,26 +93,120 @@ func NewArrowFromConn(driverConn driver.Conn) (*Arrow, error) {
 	return &Arrow{conn: conn}, nil
 }
 
-var _ array.RecordReader = (*arrowStreamReader)(nil)
-
-// arrowStreamReader implements array.RecordReader for streaming DuckDB results.
-type arrowStreamReader struct {
-	ctx      context.Context
-	res      *arrowmapping.Arrow
-	schema   *arrow.Schema
-	rowCount uint64
-
-	mu         sync.Mutex // protects readCount and currentRec
-	refCount   int64
-	readCount  uint64
-	currentRec arrow.Record
-	closed     bool // tracks if the reader has been closed/released
-	err        error
+// QueryContext prepares statements, executes them, returns Apache Arrow array.RecordReader as a result of the last
+// executed statement. Arguments are bound to the last statement.
+func (a *Arrow) QueryContext(ctx context.Context, query string, args ...any) (array.RecordReader, error) {
+	if a.conn.closed {
+		return nil, errClosedCon
+	}
+
+	rows, err := a.conn.QueryContext(ctx, query, a.anyArgsToNamedArgs(args))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Should we call `rows.Close()` here before returning the error?

Reply:
It's still using in the record reader and will be closed by last recordReader.Release. So it should be in non-closed state here